### PR TITLE
Fix some analyzer and linter warnings.

### PR DESCRIPTION
**Pull request by jeffscaturro-aka**
**Why?**
---
We had just cleaned up our project to contain zero analyzer and linter warnings after updating to the latest version of Flutter. After running code generation for some queries and mutations that we started porting over to utilize Artemis, we introduced quite a bit of analyzer and linter warnings. This didn't feel good, especially because these seemed pretty minor.

In addition, our CI caught that some files that needed formatting, and they were from the library forwarder generated files.

This meant we'd have to run code generation, and then clean up some of these files manually which felt counterintuitive, especially seeing as we run other code generation that's not related to GQL. Then we'd have to do this every time we ran code generation.

Screen Shot 2020-08-13 at 4 08 02 PM

**What?**
---
Separating this out into three issues:

1. `Annotate overridden members.`:
It looked like we can add the annotation to the `fromJson` and `toJson` methods that override the `JsonSerializable`'s implementation in `generateArgumentClassSpec`.

2. `Unused import: 'package:meta/meta.dart'. Try removing the import directive.`:
This one looked a little trickier. After looking, it looks like we add the `required` annotation only if there is a `ClassDefinition` that's an input _and_ contains a non-null property, or if it's a mutation argument class with a non-null property. 
The gap here looked like we checked if the `ClassDefinition` was an input class when adding the annotation, but when checking to see if we should add the `meta` package (required for that annotation) we didn't check if it was an input class. 

3. Use single quotes instead of double quotes in `JsonValue`.

## Notes

I went ahead and bumped the version, and added a changelog entry for this version. Let me know if you'd like me to revert that if you intend to bundle this up with another change.

--- 
Let me know if you'd like to see any other changes, or if there are some hidden side-effects these changes would have. These changes seemed pretty straightforward and low-impact.

### CODE REVIEW

File: lib/generator/helpers.dart [hunk 1]
```diff
@@ -37,21 +37,21 @@ extension ExtensionsOnIterable<T, U> on Iterable<T> {
...
-    for (final input in query.inputs) {
-      if (input.isNonNull) {
+      if (clazz.isInput && clazz.properties.any((p) => p.isNonNull)) {
         return true;
       }
     }
+
+    if (query.inputs.any((i) => i.isNonNull)) {
+      return true;
+    }
```
Comment (jeffscaturro-aka) [inline lib/generator/helpers.dart:54]: info: this logic wasn't changed, just switched to use the `any` function.

File: lib/generator/helpers.dart [hunk 2]
```diff
@@ -37,21 +37,21 @@ extension ExtensionsOnIterable<T, U> on Iterable<T> {
...
     for (final clazz in query.classes.whereType<ClassDefinition>()) {
-      for (final property in clazz.properties) {
-        if (property.isNonNull) {
-          return true;
-        }
-      }
-    }
-    for (final input in query.inputs) {
-      if (input.isNonNull) {
+      if (clazz.isInput && clazz.properties.any((p) => p.isNonNull)) {
```
Comment (jeffscaturro-aka) [inline lib/generator/helpers.dart:47]: info: switching the current logic to utilize the `any` function

info: **new logic** is to include `clazz.isInput`

File: lib/generator/print_helpers.dart [hunk 3]
```diff
@@ -376,4 +378,5 @@ void writeLibraryDefinitionToBuffer(
 /// a breaking change on file generation.
 String writeLibraryForwarder(LibraryDefinition definition) =>
     '''// GENERATED CODE - DO NOT MODIFY BY HAND
-export '${definition.basename}.dart';''';
+export '${definition.basename}.dart';
+''';
```
Comment (jeffscaturro-aka) [inline lib/generator/print_helpers.dart:382]: info: new line so `flutter format` doesn't have to add it

File: lib/generator/print_helpers.dart [hunk 4]
```diff
@@ -210,13 +210,15 @@ Spec generateArgumentClassSpec(QueryDefinition definition) {
...
           ))
+          ..annotations.add(CodeExpression(Code('override')))
           ..body = Code('_\$${definition.className}ArgumentsFromJson(json)'),
       ))
       ..methods.add(Method(
         (m) => m
           ..name = 'toJson'
           ..lambda = true
           ..returns = refer('Map<String, dynamic>')
+          ..annotations.add(CodeExpression(Code('override')))
```
Comment (jeffscaturro-aka) [inline lib/generator/print_helpers.dart:221]: info: these two methods override `JsonSerializable`'s implementation and were just missing the annotation

**Comment by jeffscaturro-aka** [comment]
For your attention: @comigor

**Comment by jeffscaturro-aka** [comment]
Not sure if CI failing is actually related to these changes, I don't see much output except for:
```
# github.com/itchyny/gojq/cli
/go/pkg/mod/github.com/itchyny/gojq@v0.7.1-0.20200105143805-d24ecb5d89a9/cli/cli.go:154:13: err.Position undefined (type participle.Error has no field or method Position)
/go/pkg/mod/github.com/itchyny/gojq@v0.7.1-0.20200105143805-d24ecb5d89a9/cli/cli.go:159:36: err.Position undefined (type participle.Error has no field or method Position)
/go/pkg/mod/github.com/itchyny/gojq@v0.7.1-0.20200105143805-d24ecb5d89a9/cli/cli.go:160:38: err.Position undefined (type participle.Error has no field or method Position)
/go/pkg/mod/github.com/itchyny/gojq@v0.7.1-0.20200105143805-d24ecb5d89a9/cli/cli.go:164:59: err.Position undefined (type participle.Error has no field or method Position)
/go/pkg/mod/github.com/itchyny/gojq@v0.7.1-0.20200105143805-d24ecb5d89a9/cli/cli.go:165:30: err.Position undefined (type participle.Error has no field or method Position)
##[error]Process completed with exit code 2.
```

Both the version and changelog were updated.

**Comment by vasilich6107** [comment]
Hi @jeffscaturro-aka main action occurs in `beta` branch.
It would be great if you rebase your PR onto `beta`.
I would also recommend to use `beta` in your project as far as it contains a lot of bug fixes

File: lib/generator/print_helpers.dart (hunk 3 above)
Comment (jeffscaturro-aka) [reply lib/generator/print_helpers.dart]: Looks like this was already in `beta`, resolving.

**Comment by vasilich6107** [comment]
Hi @jeffscaturro-aka I'm testing your PR.
It adds a bunch of `@overrrides` but I do not see that this changes anything.
I checked the readme
https://github.com/google/json_serializable.dart/tree/master/json_serializable
and exmaple
https://github.com/google/json_serializable.dart/blob/master/json_serializable/example/example.dart 

I do not see the need of `@overrides`

Is your case relates to some `analyzer` rule that produces this type of warnings?

**Comment by vasilich6107** [comment]
It seems that you are using `annotate_overrides` https://dart-lang.github.io/linter/lints/annotate_overrides.html

File: test/generator/print_helpers_test.dart [hunk 5]
```diff
@@ -47,13 +47,13 @@ void main() {
       final str = specToString(enumDefinitionToSpec(definition));
 
       expect(str, '''enum Name {
-  @JsonValue("Option")
+  @JsonValue('Option')
```
Comment (vasilich6107) [inline test/generator/print_helpers_test.dart:50]: What is the cause of this replacements from  `"` to `'`?

Comment (jeffs-sf) [reply test/generator/print_helpers_test.dart:50]: Dart best practices prefer the use of single quotes where they don't require additional escapes: https://dart-lang.github.io/linter/lints/prefer_single_quotes.html

Comment (jeffscaturro-aka) [reply test/generator/print_helpers_test.dart:50]: Dart best practices prefer the use of single quotes when there aren't additional escapes required: https://dart-lang.github.io/linter/lints/prefer_single_quotes.html

**Comment by jeffscaturro-aka** [comment]
Thanks for the approvals @comigor and @vasilich6107 - is there anything else you need from me in order to get this merged?

**Comment by comigor** [comment]
Nothing, I should have merged it :P

**Comment by jeffscaturro-aka** [comment]
Sweet, thank you!

**Pull request merged**